Update `pify` to v5.0.0

Gudahtt opened the pull request:
`pify` v5.0.0 will preserve `this` references correctly, so explicit binding is no longer needed.

There are no breaking changes that affect us; the only breaking change in v4 and v5 is to update the minimum Node.js version to v10.

whymarrh commented:
> so explicit binding is no longer needed.

This make sense, but, for example, in these cases here:

```diff
- return await pify(this.ethQuery.getBlockByNumber).call(this.ethQuery, blockNumberHex, true)
+ return await pify(this.ethQuery.getBlockByNumber)(blockNumberHex, true)
```

How do we know that the function is pre-bound?

Gudahtt commented:
We don't; that was a mistake on my part. I'll revert that. Good catch!

This saves us from pre-binding when we're passing an _object_ to `pify`, but we don't do that anywhere in this repo. For functions things are unchanged.

Gudahtt committed:
Update `pify` to v5.0.0

`pify` v5.0.0 will preserve `this` references correctly, so explicit
binding of objects passed to `pify` is no longer needed.

There are no breaking changes that affect us; the only breaking change
in v4 and v5 is to update the minimum Node.js version to v10.
